src/main/kotlin/org/jitsi/nlj/stats/DelayStats.kt

```diff
@@ -18,24 +18,32 @@ package org.jitsi.nlj.stats
 
 import org.jitsi.nlj.PacketInfo
 import org.jitsi.utils.stats.BucketStats
+import java.time.Clock
+import java.time.Duration
 
 open class DelayStats(thresholdsNoMax: LongArray = defaultThresholds) :
     BucketStats(thresholdsNoMax, "_delay_ms", " ms") {
 
+    fun addDelay(delay: Duration?) {
+        if (delay != null) addDelay(delay.toMillis())
+    }
     fun addDelay(delayMs: Long) = addValue(delayMs)
 
     companion object {
         val defaultThresholds = longArrayOf(2, 5, 20, 50, 200, 500, 1000)
     }
 }
 
-class PacketDelayStats(thresholdsNoMax: LongArray = defaultThresholds) : DelayStats(thresholdsNoMax) {
+class PacketDelayStats(
+    thresholdsNoMax: LongArray = defaultThresholds,
+    private val clock: Clock = Clock.systemUTC()
+) : DelayStats(thresholdsNoMax) {
     fun addPacket(packetInfo: PacketInfo) {
-        val delayMs = if (packetInfo.receivedTime > 0) {
-            System.currentTimeMillis() - packetInfo.receivedTime
+        val delay = if (packetInfo.receivedTime != null) {
+            Duration.between(packetInfo.receivedTime, clock.instant())
         } else {
-            -1
+            null
         }
-        addDelay(delayMs)
+        addDelay(delay)
```

Comment:
Maybe
```
packetInfo.receivedTime?.let { receivedTime ->
    addDelay(Duration.between(receivedTime, clock.instant())
}
```
?

Then `addDelay` doesn't have to support accepting a null value.

Reply:
Yeah, I noticed that in parallel with you.